templates/cmd/controller/main.go.tpl

```diff
@@ -102,7 +103,11 @@ func main() {
 	)
 	sc := ackrt.NewServiceController(
 		awsServiceAlias, awsServiceAPIGroup, awsServiceEndpointsID,
-		ackrt.VersionInfo{},	// TODO: populate version info
+		ackrt.VersionInfo{
+			version.GitCommit,
```

Comment:
I see these values as empty in the generated controllers right now. What am i missing?

Reply:
Yeah the LDFLAGS for individual controllers are not being loaded correctly. [You can see](https://github.com/aws-controllers-k8s/s3-controller/blob/main/Makefile#L7-L12) the version is pinned at `v0.0.0` and it's modifying variables that are not at that location (it's trying to reference them in main). Should probably model these flags after the [LDFLAGS in code-generator](https://github.com/aws-controllers-k8s/code-generator/blob/main/Makefile#L9-L15). These one load the version from the dirty git tag (which is clean when built through CI/CD), and properly use import paths for the variable replacement.

Reply:
The Makefile is not included in `code-generator`, but IMO it should be. There isn't anything in there that is service-specific anyway, so I don't see any problem with doing so.

Reply:
>The Makefile is not included in code-generator, but IMO it should be.

For the most part, yes! But a service team may modify their Makefile and we would not wanna override it from code-generator in that case. Solution for that problem can be custom hooks for adding new targets to auto-generated Makefile.

Reply:
Got it now. These variables are set at build-time.
So this PR looks good to be merged however we still need to fix Makefile to pass the correct ldflags.